Source/WebCore/Modules/webcodecs/WebCodecsAudioDecoder.cpp

```diff
@@ -250,6 +250,7 @@ ExceptionOr<void> WebCodecsAudioDecoder::resetDecoder(const Exception& exception
     if (m_state == WebCodecsCodecState::Closed)
         return Exception { ExceptionCode::InvalidStateError, "AudioDecoder is closed"_s };
 
+    Ref pendingActivityKeepingJSWrapperAlive = makePendingActivity(*this);
```

Comment:
Why is it needed here?
resetDecoder is not synchronously executing any JS so I would assume we are ok without it.

Reply:
I added it because of the state transition to `Unconfigured` which means `virtualHasPendingActivity` will not protect the object anymore. But it's also probably unnecessary like you said. I removed it and the tests are all passing with many iterations and gc.